docs/developer/app-store/apps/adyen.mdx

```diff
@@ -0,0 +1,355 @@
+
+If the `errors` field doesn't exist or is an empty array, pass the `paymentDetailsResponse` back to Adyen Drop-in. The Drop-in will handle the response and display the result to the user or again require additional actions to proceed.
+
+Repeat the step until the payment is successful or fails.
+
+:::caution
+
+Many payment methods are not settled synchronously. Sometimes it takes seconds, minutes, hours, or even days for a payment to go through. Adyen App will automatically handle Adyen webhook notifications and create transaction events in Saleor (see [`transactionEventReport`](../../../api-reference/payments/mutations/transaction-event-report)).
+
+:::
+
+### Handling errors
+
+The three mutations described above may return `data.errors` field. The existence of this field determines that the request was unsuccessful. `errors` is an array of `SyncWebhookAppError` objects. The `SyncWebhookAppError` object has the following fields:
+
+```ts
+{
+  code?: string;
+  message?: string;
+  details?: JSONObject;
+}
+```
+
+The `code` field is a string identifying the error. One of the following values is allowed:
+
+- `UnknownError`
+- `JsonSchemaError`
+- `MissingSaleorApiUrlError`
+- `MissingAuthDataError`
+- `HttpClientError`
+
+:::note
+
+This list may be extended in the future. Make sure your app handles unknown error codes.
+
+:::
+
+The `message` field is a human-readable message describing the error.
+
+The `details` field is an object containing additional information about the error. It may contain two fields:
+
+- `errorCode` – [Adyen error code](https://docs.adyen.com/development-resources/error-codes)
+- `statusCode` – [Adyen HTTP status code](https://docs.adyen.com/development-resources/response-handling)
+
+Example:
+
+```json
+{
+  "data": {
+    "transactionInitialize": {
+      "transactionEvent": {
+        "pspReference": "",
+        "amount": {
+          "amount": 54.24,
+          "currency": "EUR"
+        },
+        "type": "AUTHORIZATION_FAILURE"
+      },
+      "data": {
+        "errors": [
+          {
+            "code": "HttpClientError",
+            "message": "HTTP Exception: 422. : Unable to decrypt data",
+            "details": {
+              "errorCode": "174",
+              "statusCode": 422
+            }
+          }
+        ],
+        "paymentResponse": {}
+      },
+      "errors": []
+    }
+  }
+}
+```
+
+## Development
+
+To run the Adyen App locally:
+
+1. Go to the app directory.
+2. Copy the `.env.example` file to `.env`.The `.env` should contain the following variables:
+
+:::info
+
+Adyen App is a Next.js application. If you want to learn more about setting environment variables in Next.js, head over to the [documentation](https://nextjs.org/docs/basic-features/environment-variables).
+
+:::
```

Comment:
We have much more env's required for running tests. Maybe we should add a note about that? (`src/__tests__/test-env.mjs`)

Reply:
Then maybe we should get rid of this section altogether?